### PR TITLE
Wip/hughsie/dell fix another way

### PR DESCRIPTION
**Pull request by hughsie**
Type of pull request:

- [ ] New plugin (Please include [new plugin checklist](https://github.com/fwupd/fwupd/wiki/New-plugin-checklist))
- [X] Code fix
- [ ] Feature
- [ ] Documentation

### CODE REVIEW

**Review by CragW — APPROVED** [review]

**Comment by hughsie** [comment]
@CragW success? It seems to work on my XPS 13 here

**Review by superm1 — DISMISSED** [review]

File: plugins/synaptics-mst/fu-self-test.c [hunk 1]
```diff
@@ -34,7 +34,9 @@ _test_add_fake_devices_from_dir(FuPlugin *plugin, const gchar *path)
 	g_assert_no_error(error);
 	g_assert_nonnull(dir);
 
-	ret = fu_context_load_quirks(ctx, FU_QUIRKS_LOAD_FLAG_NO_CACHE, &error);
+	ret = fu_context_load_quirks(ctx,
+				     FU_QUIRKS_LOAD_FLAG_NO_CACHE | FU_ENGINE_LOAD_FLAG_HWINFO,
```
Comment (CragW) [inline plugins/synaptics-mst/fu-self-test.c:38]: I think we'll need the header file for `FU_ENGINE_LOAD_FLAG_HWINFO`

Comment (hughsie) [reply plugins/synaptics-mst/fu-self-test.c:38]: Doh, I'm an idiot trying to multitask :) Will fix.

**Review by CragW — APPROVED** [review]

**Comment by superm1** [comment]
```
# Start of fwupd tests
# FuPlugin-DEBUG: coldplug(dell)
# FuPluginDell-DEBUG: TPM HW version: 0x0
# FuPluginDell-DEBUG: TPM Status: 0x0
# FuPluginDell-DEBUG: no switchable TPM detected: TPM not enabled: 0x0
Bail out! FuPluginDell:ERROR:../fwupd/plugins/dell/fu-self-test.c:115:fu_dell_plugin_tpm_func: assertion failed (error == NULL): failed to query system for TPM information: 0xfffffffe (g-io-error-quark, 15)
```

The arch CI seems to point out more change needed for `plugins/dell/fu-self-test.c`

**Comment by hughsie** [comment]
> The arch CI seems to point out more change needed

On it.

**Review by superm1 — APPROVED** [review]

**Pull request merged**